gtsam/inference/DotWriter.h

```diff
@@ -35,6 +40,22 @@ struct GTSAM_EXPORT DotWriter {
                              ///< the dot of the factor
   bool binaryEdges;          ///< just use non-dotted edges for binary factors
 
+  /**
+   * Variable positions can be optionally specified and will be included in the
+   * dor file with a "!' sign, so "neato" can use it to render them.
+   */
+  std::map<Key, Vector2> variablePositions;
+
+  /**
+   * The position hints allow one to use symbol character and index to specify
```

Comment:
I like this functionality

Reply:
Do we want to specify factor positions as well?

Reply:
Yeah, forgot about that. will add. Currently in graphVizFormatting, actually, but I need it here.